src/programs/project3d/project3d.cpp

```diff
@@ -274,16 +287,50 @@ bool Project3DApp::DoCalculation( ) {
                 final_image.CosineMask(mask_radius / input_parameters.pixel_size, 6.0);
 
 #pragma omp ordered
+
```

Comment:
if current_image is sufficient to limit access to the correct thread when updating the output params, I wonder why the omp ordered clause is needed here?

Reply:
I believe this is for ensuring proper order when writing the stack, as the ordered clause applies only to ```final_image.WriteSlice(&output_image, current_image + 1);``` since there are no braces following it to include any of the following code. I think the same applies to the atomic clause, so it appears the parameters are being updated without additional clauses.

Reply:
Right, but each thread has its own value of current_image, and afaik as long as only one thread writes at a time, it would happen at the correct place in the std::fstream buffer that is managed by the MRCFile object. (It **may** be a problem in the below, but afaik the check on my_file->tellg() is an optimization to avoid extra calls to my_file->seekg which aren't needed for the in order operation. (Because after writing slice 2, my_file->tellg() should give the start of slice 3)

```cpp
MRCFile::WriteSlicesToDisk( ...
        long image_offset     = long(start_slice - 1) * bytes_per_slice;
        long current_position = my_file->tellg( );
        long seek_position    = 1024 + image_offset + long(my_header.SymmetryDataBytes( ));

        if ( current_position != seek_position )
            my_file->seekg(seek_position);
```

What I don't know, is 
a) how large the std::fstream buffer is 
and
b) how often it is flushed to disk

If it is only flushed to disk on f.close() and the writes are in bigish chunks, then a call to my_file->seekg() would [i think] just be moving a pointer in memory (to the buffer) and so is cheap compared to a seek on disk. 

None of this really matters for this PR - I just like to explore the "whys" when they come up as it is often useful later on. Continuing ed sorta thing : )